core/src/com/unciv/logic/automation/unit/WorkerAutomation.kt

```diff
@@ -105,6 +105,12 @@ class WorkerAutomation(
         result
     }
 
+    /** Cache of roads to connect cities each turn */
+    private val roadsToConnectCitiesCache: HashMap<City, List<Tile>> = HashMap()
+    
+    /** Hashmap of all cached tiles in each list in [roadsToConnectCitiesCache] */
+    private val tilesOfRoadsToConnectCities: HashMap<Tile, City> = HashMap()
```

Comment:
Surely a tile between two cities could link to two cities. In that case do you sort by closest? That's what I gather from 294 and if so that's fine

Reply:
Actually there is no reason to sort in the first place since we are now converting line 294 to a hashmap instead of a list.
As for your question, I don't exactly understand.